Repository clarification (#210)
bufbuild/docs.buf.build: docs/bsr/overview.md

- **Remote**: The DNS name for the server hosting the BSR. This is always `buf.build`.
- **Owner**: An entity that is either a user or organization within the BSR ecosystem.

> - **Repository**: Stores all the different versions of the same module. 

Reviewer: Nit suggestion: stores all versions of a module

Author: I do want to strongly emphasize that it's *different* versions of the *same* module. `Stores all versions of a module` fails to imply that a repository only stores one module

Reviewer: Hmmm perhaps, `stores all version of a single module` would address that?

> - **Repository**: Stores all the different versions of the same module. 

Reviewer: ```suggestion
- **Repository**: Stores all versions of a single module
```

## Referencing a module

> Each module in a BSR repository exists as a snapshot, and contains a unique reference associated with every change.

Reviewer: Suggestion: A module exists as snapshots in a BSR repository. A snapshot can include any number of changes, where each change is identifiable by a unique reference.

> Each module in a BSR repository exists as a snapshot, and contains a unique reference associated with every change.

Reviewer: ```suggestion
A module exists as snapshots in a BSR repository. A snapshot can include any number of changes, where each change is identifiable by a unique reference.
```

Author: imo we should ditch the use of the word "snapshot" and just use "version"

Reviewer: I agree with this but want to be consistent with the above definition. If we remove usage of `snapshot` above, then I'm down for `version` here.

Reviewer: I agree with this. There was already mixed usage in docs where folks referred to `commit` as `version`, which means we're predisposed to thinking of it as a `version`.

> Each module in a BSR repository exists as a snapshot, and contains a unique reference associated with every change.

Reviewer: ```suggestion
A module has different versions. Each version includes any number of changes, and each change is identifiable by a unique commit or tag. The collective set of module versions is housed within a repository.
```

> A reference is a way to refer to a single version of the module. While a reference always _resolves_ to a single version of the module, it can be either a commit or a tag.

Reviewer: (If the above suggestion is accepted), remove `A reference is a way to refer to a single version of the module`, then:

```
While a reference always _resolves_ to a single snapshot, it can be either a commit or a tag.
```

> A reference is a way to refer to a single version of the module. While a reference always _resolves_ to a single version of the module, it can be either a commit or a tag.

Reviewer: ```suggestion
While a reference _always_ resolves to a single snapshot, it can be either a commit or a tag.
```

> A reference is a way to refer to a single version of the module. While a reference always _resolves_ to a single version of the module, it can be either a commit or a tag.

Reviewer: ```suggestion
```
